base-compat.cabal

```diff
@@ -22,8 +23,12 @@ library
       -Wall
   build-depends:
       base == 4.*
-    , ghc-prim
     , setenv
+
+  if impl(ghc >= 7.2) && impl(ghc < 7.6)
+    -- Prior to GHC 7.6, GHC.Generics lived in ghc-prim
+    build-depends: ghc-prim
```

Comment:
I added it since the `GHC.Generics` module was a part of `ghc-prim` for precisely those versions of GHC, so I wanted to avoid needlessly depending on the package if it's not necessary. I was also implementing [a suggestion](https://github.com/sol/base-compat/pull/9#discussion_r27785189) you gave on the last PR, although I might have misunderstood what you meant.